adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/utils/GeodeticUtils.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright (c) 2020. MobilityData IO.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.utils;
+
+import org.locationtech.spatial4j.distance.DistanceCalculator;
+import org.locationtech.spatial4j.distance.DistanceUtils;
+import org.locationtech.spatial4j.shape.Point;
+import org.locationtech.spatial4j.shape.ShapeFactory;
+
+import static org.locationtech.spatial4j.context.SpatialContext.GEO;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.ShapePoint;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceCalculationUtils;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceUnit;
+
+import java.util.Map;
+
+/**
+ * Interface implementation using external library org.locationtech.spatial4j. This is used to compute distance between
+ * two {@code ShapePoint}.
+ */
+public class GeodeticUtils implements DistanceCalculationUtils {
+
+    /**
+     * Return the distance between two points given there lat/lon positions in the specified unit. Note that points of
+     * origin and destination can be swapped.
+     *
+     * @param originLatitude        latitude of the origin point
+     * @param originLongitude       longitude of the origin point
+     * @param destinationLatitude   latitude of the destination point
+     * @param destinationLongitude  longitude of the destination point
+     * @param distanceUnit          unit of the desired result of computation
+     * @return the distance between two points given there lat/lon positions in the specified unit
+     */
+    @Override
+    public double distanceBetweenTwoPoints(final float originLatitude, final float originLongitude,
+                                           final float destinationLatitude, final float destinationLongitude,
+                                                  final DistanceUnit distanceUnit) {
+        final ShapeFactory shapeFactory= getShapeFactory();
+        final DistanceCalculator distanceCalculator = getDistanceCalculator();
+
+        final Point origin = shapeFactory.pointXY(originLongitude, originLatitude);
+        final Point destination = shapeFactory.pointXY(destinationLongitude, destinationLatitude);
+        // implementation uses harvesine formula which determines the great-circle distance between two points on a
```

Comment:
You should add this to the Javadoc for the method too - it's important to know how the distance is being calculated.

Also, just curious - how did you know this? I looked at the Javadocs for Spatial4j but didn't see the default distance implementation. It would be good to include a link so someone can look more info if needed.

Reply:
It's  not straighforward from the documentation, however I did find this resources (https://locationtech.github.io/spatial4j/apidocs/org/locationtech/spatial4j/context/SpatialContext.html, section `Constructor Detail - SpatialContext`) that specifies the default implementation

Reply:
`SpatialContext` constructor isn't being used here though, right? Those methods are deprecated. You're just referencing the static `GEO.getDistCalc()`. And [those docs](https://locationtech.github.io/spatial4j/apidocs/org/locationtech/spatial4j/context/SpatialContext.html#GEO) just say:

>A popular default SpatialContext implementation for geospatial.

But poking around in [`SpatialContext`](https://github.com/locationtech/spatial4j/blob/63b318368ce7eca0e090af64cd49d134c8619c29/src/main/java/org/locationtech/spatial4j/context/SpatialContext.java#L159) and [`SpatialContextFactory`](https://github.com/locationtech/spatial4j/blob/63b318368ce7eca0e090af64cd49d134c8619c29/src/main/java/org/locationtech/spatial4j/context/SpatialContextFactory.java#L167) source code confirms you're correct - the default is haversine because `isGeo()` is `true`.

Also, note there is a typo in the comment above - `harvesine` -> `haversine`.